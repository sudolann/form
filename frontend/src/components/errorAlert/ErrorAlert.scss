p.alert {
    color: red;
    font-size: 0.8rem;
    text-align: left;
}

.flex {
    p {
        font-size: 2rem;
    }
}