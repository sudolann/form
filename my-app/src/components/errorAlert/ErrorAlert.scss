p.alert {
    color: red;
    font-size: 0.8rem;
}