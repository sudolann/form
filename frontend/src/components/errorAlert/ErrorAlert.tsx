import React, { FunctionComponent, ReactElement } from 'react';
import './ErrorAlert.scss';

export const ErrorAlert: FunctionComponent<{
  errorMessage: string;
  fullPage?: boolean;
}> = ({ errorMessage, fullPage }): ReactElement => {
  if (fullPage) {
    return (
      <div className="flex" data-testid="error">
        <p className="alert">{errorMessage}</p>
      </div>
    );
  }

  return (
    <p className="alert" data-testid="error">
      {errorMessage}
    </p>
  );
};
