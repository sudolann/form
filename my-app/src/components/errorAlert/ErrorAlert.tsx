import React, { FunctionComponent, ReactElement } from 'react';
import './ErrorAlert.scss';

export const ErrorAlert: FunctionComponent<{
  errorMessage: string;
}> = ({ errorMessage }): ReactElement => {
  return (
    <div>
      <p className="alert">{errorMessage}</p>
    </div>
  );
};
