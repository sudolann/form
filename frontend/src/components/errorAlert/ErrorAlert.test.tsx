import React from 'react';
import { ErrorAlert } from './ErrorAlert';
import '@testing-library/jest-dom/extend-expect';

import { render } from '@testing-library/react';

describe('ErrorAlert', () => {
  it('renders alert message', () => {
    const { getByText } = render(<ErrorAlert errorMessage="test" />);
    const alertText = getByText(/test/i);
    expect(alertText).toBeInTheDocument();
  });

  it('should take a snapshot', () => {
    const { asFragment } = render(<ErrorAlert errorMessage="test" />);
    expect(asFragment()).toMatchSnapshot();
  });
  it('should take a snapshot when fullPage is true', () => {
    const { asFragment } = render(
      <ErrorAlert errorMessage="test" fullPage />,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
