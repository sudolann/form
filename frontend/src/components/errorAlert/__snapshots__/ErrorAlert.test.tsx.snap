// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorAlert should take a snapshot 1`] = `
<DocumentFragment>
  <p
    class="alert"
    data-testid="error"
  >
    test
  </p>
</DocumentFragment>
`;

exports[`ErrorAlert should take a snapshot when fullPage is true 1`] = `
<DocumentFragment>
  <div
    class="flex"
    data-testid="error"
  >
    <p
      class="alert"
    >
      test
    </p>
  </div>
</DocumentFragment>
`;
